### PR TITLE
Make Type[T] comparisons work

### PR DESCRIPTION
**Pull request by Michael0x2a**
This commit introduces a workaround to fix the bug discussed in https://github.com/python/mypy/pull/1787

Previously, code where you compared two types (eg `int == int` or `int != int`) would cause mypy to incorrectly report a "too few arguments" error.

### CODE REVIEW

**Comment by gvanrossum** [comment]
@JukkaL What do you think of this? It works but it feels a bit ad-hoc. The same behavior applies to all operators (everything in nodes.op_methods and nearby variables) but it feels the key problem is happening because of some confusion between the class and the metaclass (or between the instance and the class) -- it's just only apparent for `__eq__` because that's the only interesting operator defined on both type and object. I've got a feeling that maybe there's something that check_op() or check_op_local() could do better?

**Comment by gvanrossum** [comment]
More background: for operators, the method name (in this case `__eq__`) is never looked up on the instance, only on the class; in the case of class objects, the "instance" is the class and the "class" is the metaclass. Without this rule we wouldn't be able to compare class objects using == at all, at least not for classes that override equality for their instances...

**Comment by JukkaL** [comment]
Yeah, this feels a little ad hoc. It seems that `__eq__` isn't really special -- for example, `type(1) < type('x')` works in Python 2, and arguably should be accepted by mypy. Maybe we should have the same logic for all operators if the left operand has type `Type[x]`?

File: mypy/checkmember.py [hunk 1]
```diff
@@ -91,10 +92,24 @@ def analyze_member_access(name: str, typ: Type, node: Context, is_lvalue: bool,
         if isinstance(ret_type, TupleType):
             ret_type = ret_type.fallback
         if isinstance(ret_type, Instance):
-            result = analyze_class_attribute_access(ret_type, name, node, is_lvalue,
-                                                    builtin_type, not_ready_callback, msg)
-            if result:
-                return result
+            if not isinstance(node, (OpExpr, ComparisonExpr)):
```
Comment (ddfisher) [inline mypy/checkmember.py]: `node` is just used as a context for errors, and isn't necessarily reliable. I don't think it's the best idea to use it for this purpose.

Comment (ddfisher) [inline mypy/checkmember.py]: Also, this doesn't catch all cases, e.g. __getitem__.

Comment (ddfisher) [inline mypy/checkmember.py]: I'd consider adding an optional bool parameter to this function instead and setting it in `check_op` (in `checkexpr.py`).

**Comment by ddfisher** [comment]
Can you fix the problems surfaced by CI?  This looks good otherwise.

**Comment by Michael0x2a** [comment]
Oh, whoops -- I forgot to test after merging. Doing that now.

**Pull request merged**